### ISSUE TITLE
Error when installing via Cocoapods: 'The following Swift pods cannot yet be integrated as static libraries'

### ISSUE THREAD

**Issue by paul019** [mapbox/mapbox-navigation-ios#1791]
I am trying to install `MapboxNavigation` in my React Native and iOS project using Cocoapods. My `Podfile` looks something like this:

```
source 'https://github.com/CocoaPods/Specs.git'
platform :ios, '9.0'

EXPO_CPP_HEADER_DIR = 'ExpoKit'

target 'pq-app-v2' do
  pod 'Mapbox-iOS-SDK', '~> 4.5',
    :modular_headers => true
  pod 'MapboxNavigation', '~> 0.22.0',
    :modular_headers => true
 pod 'React',
    :path => "../node_modules/react-native",
    :inhibit_warnings => true,
    :subspecs => [
      "Core",
      "ART",
      "RCTActionSheet",
      "RCTAnimation",
      "RCTCameraRoll",
      "RCTGeolocation",
      "RCTImage",
      "RCTNetwork",
      "RCTText",
      "RCTVibration",
      "RCTWebSocket",
      "DevSupport",
      "CxxBridge"
    ]
// ... (some other pods)
end
```

But when I run `pod repo update && pod install` in the console I get the following error:
```
[!] The following Swift pods cannot yet be integrated as static libraries:

The Swift pod `MapboxCoreNavigation` depends upon `MapboxMobileEvents`, which do not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.

The Swift pod `MapboxNavigation` depends upon `MapboxMobileEvents`, which do not define modules. To opt into those targets generating module maps (which is necessary to import them from Swift when building as static libraries), you may set `use_modular_headers!` globally in your Podfile, or specify `:modular_headers => true` for particular dependencies.
```

I do not understand this error, because I specified `:modular_headers => true` for the Mapbox dependencies in the Podfile (see above). How can I solve this error?

**Comment by frederoni** [comment]
@paul019 can you try this transitive dependency and report back?
```ruby
pod 'MapboxMobileEvents', :git => 'https://github.com/mapbox/mapbox-events-ios.git', :commit => "95185e1a5447504086876c4879f8625dc6419d8b"
```
I don't have a sample project that links mapbox-navigation-ios/mapbox-mobile-events statically handy.

**Comment by 1ec5** [comment]
The line above is in order to get the fix in mapbox/mapbox-events-ios#85, which has already been merged and released. So you should only have to `pod update` in order to verify whether the fix works. Please open a new issue if the issue persists.

Aside from that, RNMBGL v6._x_ depends on a very old version of the iOS map SDK. v7._x_ will likely be based on a compatible version. It’s being developed in nitaliano/react-native-mapbox-gl#1377; see https://github.com/nitaliano/react-native-mapbox-gl/issues/1238#issuecomment-461585880 for the latest progress update.
